src/main/java/org/sagebionetworks/web/server/PortalServletModule.java

```diff
@@ -71,8 +81,52 @@ public class PortalServletModule extends ServletModule {
     PortalServletModule.class.getName()
   );
 
-  @Override
-  protected void configureServlets() {
+  // ViteManifestProviderImpl requires ServletContext, which cannot be injected, so we create the singleton here.
+  private ViteManifestProvider viteManifestProviderSingleton;
+
+  private void bindDependencies() {
+    // The Rest template provider should be a singleton.
+    bind(RestTemplateProvider.class)
+      .to(RestTemplateProviderImpl.class)
+      .in(Singleton.class);
+
+    bind(LoggerProvider.class).to(LoggerProviderImpl.class);
+    bind(PropertyProvider.class).to(PropertyProviderImpl.class);
+    bind(ConfigurationProperties.class).to(ConfigurationPropertiesImpl.class);
+    bind(StackConfiguration.class).to(StackConfigurationImpl.class);
+    bind(StackEncrypter.class).to(StackEncrypterImpl.class);
+
+    // JSONObjectAdapter
+    bind(JSONObjectAdapter.class).to(JSONObjectAdapterImpl.class);
+  }
+
+  private void bindFilters() {
+    filter("/*").through(SSLFilter.class);
+    bind(SSLFilter.class).in(Singleton.class);
+
+    filter("*").through(HtmlInjectionFilter.class);
+    bind(HtmlInjectionFilter.class).in(Singleton.class);
+
+    filter("/*").through(GWTCacheControlFilter.class);
+    bind(GWTCacheControlFilter.class).in(Singleton.class);
+
+    filter("/js/*").through(GWTAllCacheFilter.class);
+    filter("/images/*").through(GWTAllCacheFilter.class);
+    filter("/research/*").through(GWTAllCacheFilter.class);
+    bind(GWTAllCacheFilter.class).in(Singleton.class);
+
+    filter("/*").through(JavaScriptContentTypeFilter.class);
+    bind(JavaScriptContentTypeFilter.class).in(Singleton.class);
+
+    filter("/*").through(HSTSFilter.class);
+    bind(HSTSFilter.class).in(Singleton.class);
+
+    filter("/*").through(CORSFilter.class);
+    bind(CORSFilter.class).in(Singleton.class);
+
+    filter("/*").through(XFrameOptionsFilter.class);
+    bind(XFrameOptionsFilter.class).in(Singleton.class);
```

Comment:
Removed filters from web.xml and added them back here in the same order. This allows us to inject dependencies into them.